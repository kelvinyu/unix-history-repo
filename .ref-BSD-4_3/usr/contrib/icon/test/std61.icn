record array(a,b,c,d,e,f,g)

procedure dummy(u,v,x,y,z)
   suspend u | v
   return x
end

procedure main()
   write("every write(\"abcdef\" ? tab(1 to 10)) ----> ",image(every write("abcdef" ? tab(1 to 10))) | "none")
   write("every write(\"abcde\" ? while move(2) ? move(1)) ----> ",image(every write("abcde" ? while move(2) ? move(1))) | "none")
   write("s := \"abcdef\" ----> ",image(s := "abcdef") | "none")
   write("s ?:= move(3) ----> ",image(s ?:= move(3)) | "none")
   write("s := \"abcdef\" ----> ",image(s := "abcdef") | "none")
   write("every write(s ?:= upto(&lcase)) ----> ",image(every write(s ?:= upto(&lcase))) | "none")
   write("s := \"this is the time to work it all out\" ----> ",image(s := "this is the time to work it all out") | "none")
   write("every write(s ? tab(find(\" \"))) ----> ",image(every write(s ? tab(find(" ")))) | "none")
   write("s := \"xxxxxx\" ----> ",image(s := "xxxxxx") | "none")
   write("every s ? write(=(\"a\" | \"x\")) ----> ",image(every s ? write(=("a" | "x"))) | "none")
   write("\"abcdef\" ? (tab(0) & (while write(move(-1)))) ----> ",image("abcdef" ? (tab(0) & (while write(move(-1))))) | "none")
   write("\"abcdef\" ? (tab(0) & (while write(move(\"a\")))) ----> ",image("abcdef" ? (tab(0) & (while write(move("a"))))) | "none")
end
