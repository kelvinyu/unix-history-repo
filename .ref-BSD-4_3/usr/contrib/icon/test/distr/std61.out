
a
ab
abc
abcd
abcde
abcdef
every write("abcdef" ? tab(1 to 10)) ----> none
every write("abcde" ? while move(2) ? move(1)) ----> none
s := "abcdef" ----> "abcdef"
s ?:= move(3) ----> "abc"
s := "abcdef" ----> "abcdef"
1
2
3
4
5
6
every write(s ?:= upto(&lcase)) ----> none
s := "this is the time to work it all out" ----> "this is the time to work it all out"
this
this is
this is the
this is the time
this is the time to
this is the time to work
this is the time to work it
this is the time to work it all
every write(s ? tab(find(" "))) ----> none
s := "xxxxxx" ----> "xxxxxx"
x
every s ? write(=("a" | "x")) ----> none
f
e
d
c
b
a
"abcdef" ? (tab(0) & (while write(move(-1)))) ----> none
Run-time error 101 at line 20 in std61.icn
integer expected
offending value: "a"
